AirCasting/SDSync/FixedSessions/SDCardFixedSessionsSavingService.swift

```diff
@@ -2,19 +2,90 @@
 //
 
 import Foundation
+import CoreLocation
 import Resolver
 
-//TODO: Refactor this part
-class SDCardFixedSessionsSavingService {
+enum UploadingError: Error {
+    case uploadError
+}
+
+class SDCardFixedSessionsUploadingService {
+    @Injected private var fileLineReader: FileLineReader
+    @Injected private var measurementStreamStorage: MeasurementStreamStorage
     @Injected private var apiService: UploadFixedSessionAPIService
+    private let parser = SDCardMeasurementsParser()
     
-    func processAndSync(csvSession: CSVStreamsWithMeasurements, deviceID: String, completion: @escaping (Bool) -> Void) {
-        guard !csvSession.sessions.isEmpty else {
-            completion(true)
-            return
+    private let bufferThreshold = 200
+    
+    func processAndUpload(fileURL: URL, deviceID: String, completion: @escaping (Result<[SessionUUID], Error>) -> Void) {
+        var processedSessions = Set<SessionUUID>()
+        var streamsWithMeasurements: [SDStream: [Measurement]] = [:]
+        
+        // We don't want to upload sessions of other users
+        var sessionsToIgnore: [SessionUUID] = []
+        
+        var readLines = 0
+        
+        measurementStreamStorage.accessStorage { storage in
+            do {
+                try self.fileLineReader.readLines(of: fileURL, progress: { line in
+                    switch line {
+                    case .line(let content):
+                        readLines += 1
+                        Log.info("LINE: \(content)")
+                        let measurementsRow = self.parser.parseMeasurement(lineString: content)
+                        guard let measurements = measurementsRow, !sessionsToIgnore.contains(measurements.sessionUUID) else { return }
+                        
+                        let session = processedSessions.first(where: { $0 == measurements.sessionUUID })
```

Comment:
won't it be easier and faster to make `processedSessions` a set instead of an array?

Edit: I see you used Set, kudos. Use `processedSessions.contains(measurements.sessionUUID)` then plz (it will be faster and easier to read)